File: 2016/20/aoc.js
module.exports = {
    firewallRules: function(input) {
        const ranges = []
        input.split('\n').forEach(line => {
            const info = line.trim().split('-')
            ranges.push([parseInt(info[0]), parseInt(info[1])])
        })

        let part1 = -1
        let part2 = 0

        let value = 0
        let range = [1]

        while (value < 4294967295) {
            range = ranges.filter(range => value >= range[0] && value <= range[1])
            if (range.length == 0) {
                if (part1 == -1) { part1 = value }
                part2++
                value = value + 1
            } else {
                value = range[0][1] + 1
            }
        }

        return [part1, part2]
    }
}

File: 2021/06/aoc.py
def lanternfish(input):
    part1 = 0

    fish = list(map(int,input[0].split(',')))
    days = 80
    while days != 0:
        new = 0
        for i in range(0, len(fish)):
            if fish[i] == 0:
                fish[i] = 6
                new += 1
            else:
                fish[i] -= 1

        for i in range(0, new):
            fish.append(8)

        days -= 1
    part1 = len(fish)

    return (part1, None)
